### PR TITLE
[feat] added OnboardingGuard to ensure onboarding page is only accessible once

### PR DESCRIPTION
**Pull request by saitunguturu**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- only users without a log in the adopter_profiles table can access the page
- if they already have a log it will reroute them to the home page

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 
N/A

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- make a new user and go through onboarding OR delete the row of a user currently in the adopter_profiles table and try to access onboarding
- once onboarding is completed by a user any further attempts to access it should route to the home page

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ethan-tam33

### CODE REVIEW

File: middleware.ts [hunk 1]
```diff
@@ -1,19 +1,70 @@
...
 
 export const config = {
   matcher: [
-    /*
-     * Match all request paths except for the ones starting with:
-     * - _next/static (static files)
-     * - _next/image (image optimization files)
-     * - favicon.ico (favicon file)
-     * Feel free to modify this pattern to include more paths.
-     */
```
Comment (jinkang-0) [inline middleware.ts]: I think this comment is helpful, to clarify the regex. It's best to keep this.

File: app/layout.tsx [hunk 2]
```diff
@@ -34,6 +34,12 @@ export const metadata: Metadata = {
   description: 'Description of project',
 };
 
+{
+  /* <OnboardingGuard>
+            <ProfileProvider>{children}</ProfileProvider>
+          </OnboardingGuard> */
+}
+
```
Comment (jinkang-0) [inline app/layout.tsx]: If there's no need for this code, let's get rid of it.

File: components/onboarding/OnboardingComplete.tsx [hunk 3]
```diff
@@ -1,7 +1,37 @@
...
+
+      if (error) {
+        Logger.error(`Error updating user metadata: ${error.message}`);
+        throw error;
+      }
+
+      router.push('/');
+    } catch (error) {
+      Logger.error(`Failed to complete onboarding: ${error}`);
+      setIsUpdating(false);
```
Comment (jinkang-0) [inline components/onboarding/OnboardingComplete.tsx]: Can you help me understand why we're throwing and catching this error, and logging it twice?

**Comment by ethan-tam33** [comment]
@saitunguturu @jinkang-0 just pushed some changes, lmk if you guys approve and I can merge!
